### PR TITLE
Initial setup and first validators

### PR DESCRIPTION
**Pull request by PepperTeasdale**
@policygenius/developers 👀 

This is is the initial configuration of the project with the first utility functions. Still needs a README update and a few small things, but would love some eyes on the initial bit, especially the config for anything that could be improved/ simplified.

### CODE REVIEW

**Comment by j-clark** [comment]
not totally sure how this would work yet, but what do you tink about just calling `buildSchema` again?

```javascript
const { validate, warn } = buildSchema({
  fieldOne: {
    validate: [mustBeBig]
  },
  fieldTwo: {
    validate: [dateInPast]
  },
  nestedForm: buildSchema({
    nestedFieldOne: {},
    nestedFieldTwo: {}
  }),
  nestedCollection: buildSchema([{
    nestedFieldOne: {},
    nestedFieldTwo: {}
  }])
})
```

that would save us form needing a special reserved key

File: src/buildValidations.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const buildValidators = (schema, type) => (values) => {
+  const errors = {};
+  const validatedFields = Object.keys(pickBy(schema, (obj) => get(obj, `${type}.validator`)));
+
+  forEach(validatedFields, (field) => {
+    const validateCondition = schema[field][type].validateIf;
+    const isRequired = schema[field][type].required;
+
+    if (isRequired) {
+      errors[field] = 'Required';
```
Comment (PepperTeasdale) [inline src/buildValidations.js]: What kind of configuration were you thinking? Like making things dynamically required by passing a function to that option? That is definitely something I'm planning on adding.

File: build/validators/isZipCode.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
+
+exports.default = {
+  validator: function validator(_allValues, value) {
+    (0, _invariant2.default)((0, _lodash.isString)(value) || (0, _lodash.isNil)(value), 'Zip code must be a string, was passed ' + value + ', a ' + (typeof value === 'undefined' ? 'undefined' : (0, _typeof3.default)(value)));
+    return !value || /^\d{5}$/.test(value.trim());
+  },
+  errorMessage: 'Must be a valid zip code'
+};
```
Comment (PepperTeasdale) [inline build/validators/isZipCode.js]: I assumed as much, but not sure what's the convention for publishing to NPM this way. Please share any examples of that so I can use.

**Comment by PepperTeasdale** [comment]
@j-clark Using `buildSchema` for the nested forms is something I considered. The downside would be that it wouldn't be straightforward to access the fields a level up when running validators. This is not currently supported in this implementation, but something I could see us wanting to support in the future.

As for `_fieldArrays`, I think that's a little awkward, but that's trying to mirror the redux-form terminology. Even if you have only one entry in the nested form, redux form is still going to validate like there is an array (see the `validate.js` in [this example](http://redux-form.com/6.6.3/examples/fieldArrays/)). Some other options I considered for circumventing the reserved keyword were an array value signified, but I thought that might be a little opaque.

**Pull request merged**